### PR TITLE
Banner hiding in Oscar.jl, Polymake.jl, etc

### PR DESCRIPTION
**Pull request by lkastner**

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      length(bt) == 0 || (length(bt) == 1 && length(only(bt)[1].linfo.specTypes.parameters) < 4)
+
+    # Respect the -q flag
+    isquiet = Bool(Base.JLOptions().quiet)
+    show_banner = !isquiet && isinteractive_manual && isinteractive() &&
+        Base.JLOptions().banner != 0
+    return show_banner
+end
+
+export should_show_banner
```
Comment (lgoettgens) [inline src/utils.jl:24]: I don't think this should be exported
```suggestion
```

**Review by fingolfin — APPROVED** [review]
Thanks!

I would also tend to say we should not export the new `should_show_banner`. Other than that, let's merge it?

**Comment by fingolfin** [comment]
If we have a quick fix, sure. Otherwise, I'd say let's merge this ASAP so that we can then update the other packages to use it. And then we can still update it here, and the other packages benefit "for free" (that's the whole idea of this, after all)

**Comment by lgoettgens** [comment]
I would prefer to have the banner tests here instead of in https://github.com/oscar-system/Oscar.jl/pull/3541. Otherwise, merging a fix here for julia 1.11 will break the Oscar testsuite.

**Comment by lgoettgens** [comment]
Unfortunately there are failures with regard to the banner tests

File: test/utils/Banners/banners.jl [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Pkg.develop(path="$modcdir");
+"""
+out, err = run_repl_code(setup_cmd, td)
+
+if VERSION>=v"1.9"
+    @testset "Banners" begin
+        # Banner of ModA shows
+        out, err = run_repl_code("using ModA;", td)
+        res = @test strip(out) == "Banner of ModA" broken=VERSION>=v"1.11.0-DEV"
+        if res == Test.Fail
```
Comment (lgoettgens) [inline test/utils/Banners/banners.jl:40]:
```suggestion
        if res isa Test.Fail
```

File: test/utils/Banners/banners.jl [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        res = @test strip(out) == "Banner of ModB" broken=VERSION>=v"1.11.0-DEV"
+        if res == Test.Fail
+            println("out\n$out")
+            println("err\n$err")
+        end
+
+        # Banner does not show when our module is a dependency
+        out, err = run_repl_code("using ModC;", td)
+        res = @test strip(out) == "" broken=VERSION>=v"1.11.0-DEV"
+        if res == Test.Fail
```
Comment (lgoettgens) [inline test/utils/Banners/banners.jl:65]:
```suggestion
        if res isa Test.Fail
```

**Comment by lkastner** [comment]
This can be merged. As I understand it, these required tests that are missing don't exist in that form due to the branch names.

**Comment by lgoettgens** [comment]
> This can be merged. As I understand it, these required tests that are missing don't exist in that form due to the branch names.

Yeah. @fingolfin @thofma can you force-merge this?

**Pull request merged**